### PR TITLE
Changed Chart to add auto width and gap

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Changed Chart to add auto width and gap

#### Where should the reviewer start?

Chart/doc.js

#### What testing has been done on this PR?

storybook

#### How should this be manually tested?

storybook

#### Do the grommet docs need to be updated?

auto?

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

File: src/js/components/Chart/README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-## Chart
+## undefined
```
Comment (ShimiSun) [inline src/js/components/Chart/README.md]: Doesn't look right

File: src/js/components/Chart/README.md [hunk 2]
```diff
@@ -204,6 +220,7 @@ full
     large
     xlarge
     full
+    auto
```
Comment (ShimiSun) [inline src/js/components/Chart/README.md]: Why aren't we allowing `auto` for `height` as well as for `width`?

File: src/js/components/__tests__/__snapshots__/README-test.js.snap [hunk 3]
```diff
@@ -2762,7 +2762,7 @@ Defaults to
   }
 \`\`\`
 ",
-  "Chart": "## Chart
+  "Chart": "## undefined
```
Comment (ShimiSun) [inline src/js/components/__tests__/__snapshots__/README-test.js.snap]: Probably has the same root cause as the README changes

File: src/js/components/__tests__/__snapshots__/components-test.js.snap [hunk 4]
```diff
@@ -1431,7 +1431,7 @@ selectors",
       },
     ],
     "description": "A graphical chart.",
-    "name": "Chart",
+    "name": undefined,
```
Comment (ShimiSun) [inline src/js/components/__tests__/__snapshots__/components-test.js.snap]: Probably has the same root cause as the README changes

File: src/js/components/Chart/README.md (hunk 2 above)
Comment (ericsoderberghp) [reply src/js/components/Chart/README.md]: There's nothing to automatically set the height from. Width comes from the number of values + thickness + gap.

File: src/js/components/Chart/README.md (hunk 1 above)
Comment (ericsoderberghp) [reply src/js/components/Chart/README.md]: fixed

File: src/js/components/__tests__/__snapshots__/README-test.js.snap (hunk 3 above)
Comment (ericsoderberghp) [reply src/js/components/__tests__/__snapshots__/README-test.js.snap]: fixed

File: src/js/components/__tests__/__snapshots__/components-test.js.snap (hunk 4 above)
Comment (ericsoderberghp) [reply src/js/components/__tests__/__snapshots__/components-test.js.snap]: fixed

**Comment by ericsoderberghp** [comment]
Good catch. Fixed.

**Comment by ShimiSun** [comment]
Pending on #3462

**Pull request merged**